server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -1189,24 +1190,30 @@ private CanMatchResponse canMatch(ShardSearchRequest request, boolean checkRefre
             try (canMatchSearcher) {
                 QueryShardContext context = indexService.newQueryShardContext(request.shardId().id(), canMatchSearcher,
                     request::nowInMillis, request.getClusterAlias(), request.getRuntimeMappings());
-                Rewriteable.rewrite(request.getRewriteable(), context, false);
-                final boolean aliasFilterCanMatch = request.getAliasFilter()
-                    .getQueryBuilder() instanceof MatchNoneQueryBuilder == false;
                 FieldSortBuilder sortBuilder = FieldSortBuilder.getPrimaryFieldSortOrNull(request.source());
```

Comment:
That's ok, we extract the min/max value here so the sort rewrite doesn't change anything. However we could first do the rewrite and then extract the min/max values but only if `canMatch || hasRefreshPending`. Otherwise returning a `null` min/max value should be ok.

Reply:
Tackled in https://github.com/elastic/elasticsearch/pull/65583/commits/bcf5632430a139d1b218253794df8d92b5954ba3